engines/groovie/video/roq.cpp

```diff
@@ -692,7 +691,9 @@ bool ROQPlayer::processBlockQuadVector(ROQBlockHeader &blockHeader) {
 	}
 
 	// HACK: Skip the remaining bytes
-	int32 skipBytes = endpos -_file->pos();
+	int64 skipBytes = endpos -_file->pos();
+	if (abs(skipBytes) > 2)
+		return false;
```

Comment:
any idea if skipBytes is ever 0 or 1 here? below we have

```
		if (skipBytes != 2) {
			warning("Groovie::ROQ: Skipped %d bytes", skipBytes);
		}
```

maybe we should move the warning up here and return false whenever skipBytes != 2

Reply:
If I understand it correct skipBytes equal to 0 or 2 in normal case after traverse. It can be done like you propose (move warning up and return false in this case), but I also don't know if there is normal situation when skipBytes can be 1 at least. I've finished the game with this fix and there weren't any issues with this check/return that lead to crash (may be this case was only one). The crash issue in the begining of pyramid puzzle in RU localization (save file is provided [here](https://bugs.scummvm.org/ticket/13832)) due to end assertion fail in seek when after traverse _file->pos() didn't changed, so skipBytes are equal to blockHeader.size.

So, when after traverse our _file->pos() is changed we need skip rest bytes (_skipBytes) here by this logic to make it equal to end otherwise we need skip bytes somehow and avoid assertion failure like in this case with pyramid puzzle. I've just realized that may be appropriate check on wrong situation with skipBytes would be the following (it will be more correct in this case to detect this situation when traverse didn't change _file->pos()):

```
...
	int64 skipBytes = endpos -_file->pos();
	if (skipBytes > 0) {
		if (skipBytes == blockHeader.size)
			return false;
		_file->skip(skipBytes);
		if (skipBytes != 2) {
			warning("Groovie::ROQ: Skipped %d bytes", skipBytes);
		}
	}
...
```
I can make this change if you approve it.

As to warning when skipBytes != 2, may be it can be deleted after some time due to useless if there won't be detected other skipBytes value which would affect on something after skip :).

Reply:
maybe instead of `if (skipBytes == blockHeader.size)` it should be `>=` ?

Reply:
@Die4Ever For this particular case (puzzle issue - when it is equal) >= will work fine, but I don't know if it will crash on end assertion or not after returning if we've got skipBytes > blockHeader.size..... I think it can crash due to we will skip not the rest bytes but pos of block then as skipping offset here:
`
	if (endpos != _file->pos()) {
		warning("Groovie::ROQ: BLOCK %04x Should have ended at %d, and has ended at %d", blockHeader.type, endpos, (int)_file->pos());
		warning("Ensure you've copied the files correctly according to the wiki.");
		_file->seek(MIN(_file->pos(), endpos));
	}
`
So, the only way to be sure we are not failed in this when _file->pos() didn't changed (i.e. skipBytes == blockHeader.size). You can correct me if I'm wrong.

Btw, I've realized now during debugging this issue once again, that after traverse block code we've got _file->eos() == 1 (it is ended), but skipBytes > 0.... so may be the final version of checking will be this:
`
...
	if (skipBytes > 0) {
		if (_file->eos()) 
			return false;
		_file->skip(skipBytes);
		if (skipBytes != 2)
			warning("Groovie::ROQ: Skipped %d bytes", skipBytes);
	}
...
`
What do you think?

Reply:
That last one seems good

I do wonder if the file object has a way to check if a seek is gonna work or a non-crashing seek function

Reply:
OK. I've made last commit. This should works fine.